test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/AppTests.kt

```diff
@@ -90,12 +90,34 @@ class AppTests {
         }
     }
 
+    // Check that all auth providers throw the same exception for when invalid credentials are
+    // presented.
+    @Suppress("LoopWithTooManyJumpStatements")
     @Test
-    fun login_InvalidUserThrows() = runBlocking {
-        assertFailsWith<AppException> {
-            app.login(Credentials.emailPassword("foo", "bar"))
-        }.let { exception: AppException ->
-            assertTrue(exception.message!!.startsWith("invalid username/password [error_category=3, error_code=50, link_to_server_logs="))
+    fun login_invalidCredentialsThrows() = runBlocking {
+        for (provider in AuthenticationProvider.values()) {
+            try {
+                when (provider) {
```

Comment:
Maybe rework into a `when`-block that returns a credentials and then 
```
   ?.let { 
      assertFailsWithMessage(...) {
         app.login(it)
      }
   }
```
to also assert the message 🤷